metrics: etcd_object_counts->apiserver_storage_object_counts(stable)

erain opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

https://github.com/kubernetes/kubernetes/issues/98270#issue-791357977

#### Which issue(s) this PR fixes:

Fixes #98270

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note
Rename metrics `etcd_object_counts` to `apiserver_storage_object_counts` and mark it as stable. The original `etcd_object_counts` metrics name is marked as "Deprecated" and will be removed in the future. 
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:


<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1209-metrics-stability

```

erain committed:
metrics: promote apiserver_storage_object_counts to stable

- rename etcd_object_counts to apiserver_storage_object_counts

CONTEXT https://github.com/kubernetes/kubernetes/issues/98270
FIXES https://github.com/kubernetes/kubernetes/issues/98270

erain commented:
/assign @logicalhan

erain commented:
/sig instrumentation api-machinery

logicalhan commented on staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go:
```diff
@@ -43,14 +43,22 @@ var (
 		&compbasemetrics.GaugeOpts{
 			Name:           "etcd_object_counts",
-			Help:           "Number of stored objects at the time of last check split by kind.",
+			Help:           "(Deprecated) Number of stored objects at the time of last check split by kind. This metric is replaced by apiserver_storage_object_counts.",
```
Oh you don't actually need to include the 'Deprecated' part in the help text. 
You can use: `				DeprecatedVersion: "1.21.0",` in the GaugeOpts directly.

erain replied:
Done

logicalhan commented on staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go:
```diff
@@ -77,7 +85,7 @@ var (
 	)
 )
-
+etcdObjectCounts
```
is this intentional?

erain replied:
A typo. Removed.

logicalhan commented on staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go:
```diff
@@ -86,15 +94,17 @@ func Register() {
 		legacyregistry.MustRegister(etcdRequestLatency)
 		legacyregistry.MustRegister(objectCounts)
-		legacyregistry.MustRegister(dbTotalSize)
+    legacyregistry.MustRegister(etcdObjectCounts)
+    etcdObjectCountstry.MustRegister(dbTotalSize)
```
you may need to run gofmt and golint.

IIRC: 
```shell
./hack/update_gofmt.sh
./hack/update_golint.sh
```

erain replied:
Done

erain commented:
Thanks for the review @logicalhan ! All comments addressed.

erain commented:
/retest

logicalhan commented on staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go:
```diff
@@ -86,15 +95,17 @@ func Register() {
 		legacyregistry.MustRegister(objectCounts)
-		legacyregistry.MustRegister(dbTotalSize)
+		legacyregistry.MustRegister(etcdObjectCounts)
+		etcdObjectCountstry.MustRegister(dbTotalSize)
```
typo: should be etcdObjectCounts

logicalhan replied:
ou actually this should be legacyregistry

erain replied:
Fixed and thanks! 

I just noticed the same thing from the failing tests... No idea why it got in.

caesarxuchao commented:
/triage accepted

logicalhan commented:
@brancz wants this to be named `apiserver_storage_objects` but actually I am okay with `apiserver_storage_object_counts` as a convention for gauge counters. 

/hold
(wait on @brancz's reply)

logicalhan commented:
@erain, can you amend the commit message? It seems that `#` is forbidden in commit messages.

erain commented:
> @erain, can you amend the commit message? It seems that `#` is forbidden in commit messages.

Thanks. I've updated the commit message and also rebased on master. (It seems another stable metrics got merged before this one :D)

logicalhan reviewed (approved):
/lgtm
/approve
/hold

logicalhan commented:
/priority important-soon
/hold cancel

logicalhan reviewed (approved):
/lgtm
